python/pipeline/utils/mask_classification.py

```diff
@@ -60,4 +61,219 @@ def on_button(event):
+        plot_mask = np.array(mask[il:ih, jl:jh])
+
+        for ax,template in zip(axes[0][:6], [plot_mask, template1, template2-template1,
+                                             template1, template4, template3, template3*template1]):
+            ax.matshow(template[il:ih, jl:jh], cmap=cm.get_cmap('gray'))
+            ax.contour(plot_mask, np.percentile(mask[mask>0], threshold), linewidths=0.8, colors='w')
+            ax.contour(plot_mask, [0.01], linewidths=0.8, colors='w')
+            ax.set_aspect(1)
+            ax.axis('off')
+
+        for ax,template in zip(axes[1], template5):
+            ax.matshow(template[il:ih, jl:jh], cmap=cm.get_cmap('gray'))
+            ax.contour(plot_mask, np.percentile(mask[mask > 0], threshold), linewidths=0.8, colors='w')
+            ax.contour(plot_mask, [0.01], linewidths=0.8, colors='w')
+            ax.set_aspect(1)
+            ax.axis('off')
+
+        filt_trace = signal.medfilt(trace1, window)
+        idx = detect_peaks(filt_trace, mpd=len(trace1)/window)
+        centers = np.flip(sorted(np.stack([idx, filt_trace[idx]]).T, key=lambda x: x[1]))[:7]
+        for ax,center in zip(axes[3], sorted(centers, key=lambda x: x[0])[:][0]):
+            frame = np.max(movie[0][:, :, int(center-window/2):int(center+window/2+.5)])
+            ax.matshow(frame[il:ih, jl:jh], cmap=cm.get_cmap('gray'))
+            ax.contour(plot_mask, np.percentile(mask[mask > 0], threshold), linewidths=0.8, colors='w')
+            ax.contour(plot_mask, [0.01], linewidths=0.8, colors='w')
+            ax.set_aspect(1)
+            ax.axis('off')
+
+        trace1_ax = plt.subplot(8, 1, 5)
+        trace1_ax.plot(trace1)
+        trace1_ax.plot(centers, trace1[[int(center) for center in centers]], 'or')
+
+        trace2_ax = plt.subplot(8, 1, 6)
+        trace2_ax.plot(trace2)
+
+        fig.tight_layout()
+        fig.canvas.manager.window.wm_geometry("+250+250")
+        fig.suptitle('S(o)ma, A(x)on, (D)endrite, (N)europil, (A)rtifact or (U)nknown?')
+
+        def on_button(event):
+            if event.key == 'o':
+                mask_types.append('soma')
+                plt.close(fig)
+            elif event.key == 'x':
+                mask_types.append('axon')
+                plt.close(fig)
+            elif event.key == 'd':
+                mask_types.append('dendrite')
+                plt.close(fig)
+            elif event.key == 'n':
+                mask_types.append('neuropil')
+                plt.close(fig)
+            elif event.key == 'a':
+                mask_types.append('artifact')
+                plt.close(fig)
+            elif event.key == 'u':
+                mask_types.append('unknown')
+                plt.close(fig)
+
+        fig.canvas.mpl_connect('key_press_event', on_button)
+
+        plt.show()
+
+    sns.reset_orig()
+
+    return mask_types
+
+
+def detect_peaks(x, mph=None, mpd=1, threshold=0, edge='rising',
```

Comment:
is this the same as the one in stimline. Maybe add it in pipeline/utils/signal.py and import it in stimline (as `from pipeline.utils import signal; signal.detect_peaks` rather than have it in both places.

Reply:
Also there is a find_peaks in signal.py that may be enough for whatever you are doing (or maybe not)

Reply:
will delete from stimline, also is still work in progress